src/app/core/services/rundown-timing-context-state.service.ts

```diff
@@ -0,0 +1,187 @@
+    return new BehaviorSubject(this.createEmptyRundownTimingContext())
+  }
+
+  private createEmptyRundownTimingContext(): RundownTimingContext {
+    return {
+      currentEpochTime: Date.now(),
+      durationInMsSpentInOnAirSegment: 0,
+      expectedDurationInMsForRundown: 0,
+      expectedEndEpochTimeForRundown: 0,
+      expectedStartEpochTimeForRundown: 0,
+      playedDurationInMsForOnAirPart: 0,
+      playedDurationInMsForOnAirSegment: 0,
+      remainingDurationInMsForRundown: 0,
+      expectedDurationsInMsForSegments: {},
+      startOffsetsInMsFromNextCursorForSegments: {},
+    }
+  }
+
+  private onRundownChanged(rundown: Rundown): void {
+    const rundownTimingContextSubject: BehaviorSubject<RundownTimingContext> | undefined = this.getRundownTimingContextSubject(rundown.id)
+    if (!rundownTimingContextSubject) {
+      return
+    }
+    const onAirSegment: Segment | undefined = rundown.segments.find(segment => segment.isOnAir)
+    const currentEpochTime: number = Date.now()
+    const expectedDurationsInMsForSegments: Record<string, number> = this.rundownTimingService.getExpectedDurationInMsForSegments(rundown)
+    const expectedDurationInMsForRundown: number = this.rundownTimingService.getExpectedDurationInMsForRundown(rundown, expectedDurationsInMsForSegments)
+    const expectedStartEpochTimeForRundown: number = this.rundownTimingService.getExpectedStartEpochTimeForRundown(rundown, expectedDurationInMsForRundown, currentEpochTime)
+    const expectedEndEpochTimeForRundown: number = this.rundownTimingService.getExpectedEndEpochTimeForRundown(rundown, expectedDurationInMsForRundown, currentEpochTime)
+    const playedDurationInMsForOnAirPart: number = this.rundownTimingService.getPlayedDurationInMsForOnAirPart(rundown, currentEpochTime)
+    const playedDurationInMsForOnAirSegment: number = this.rundownTimingService.getPlayedDurationInMsForOnAirSegment(rundown, currentEpochTime)
+    const durationInMsSpentInOnAirSegment: number = this.rundownTimingService.getDurationInMsSpentInOnAirSegment(rundown, currentEpochTime)
+    const remainingDurationInMsForRundown: number = this.rundownTimingService.getRemainingDurationInMsForRundown(rundown, expectedDurationsInMsForSegments, playedDurationInMsForOnAirSegment)
+    const rundownTimingContext: RundownTimingContext = {
+      onAirSegmentId: onAirSegment?.id,
+      currentEpochTime,
+      expectedDurationInMsForRundown: expectedDurationInMsForRundown,
+      expectedStartEpochTimeForRundown: expectedStartEpochTimeForRundown,
+      expectedEndEpochTimeForRundown: expectedEndEpochTimeForRundown,
+      remainingDurationInMsForRundown: remainingDurationInMsForRundown,
+      playedDurationInMsForOnAirSegment,
+      durationInMsSpentInOnAirSegment,
+      playedDurationInMsForOnAirPart,
+      expectedDurationsInMsForSegments,
+      startOffsetsInMsFromNextCursorForSegments: {},
+    }
```

Comment:
This is close to be a duplication of a good chunk of `onTimeTick()`. My comment there would also work for this.

Reply:
See https://github.com/tv2/ng-sofie/pull/67#discussion_r1401946310.